versionutils/version.go

```diff
@@ -16,10 +16,17 @@ const (
 	SemverMinimumVersion  = "v0.0.1"
 )
 
+var (
+	InvalidSemverVersionError = func(tag string) error {
+		return errors.Errorf("Tag %s is not a valid semver version, must be of the form vX.Y.Z[-rc#]", tag)
+	}
+)
+
 type Version struct {
-	Major int
-	Minor int
-	Patch int
+	Major            int
+	Minor            int
+	Patch            int
+	ReleaseCandidate int
 }
 
 func NewVersion(major, minor, patch int) *Version {
```

Comment:
nit: this could call `NewRcVersion(major, minor, patch, rc int)` with rc version 0